packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts

```diff
@@ -265,12 +265,13 @@ export class CartCheckoutPage {
 		// the "Edit" button on the payment method step. There are cases where
 		// the step will not be collapsed, however, so this will only trigger
 		// if the edit button is visible.
-		if ( await this.page.isVisible( selectors.editPaymentStep ) ) {
+		const selector = this.page.locator( selectors.existingCreditCard( cardHolderName ) ).first();
+		try {
+			await selector.click();
+		} catch ( error ) {
 			await this.page.click( selectors.editPaymentStep );
+			await selector.click();
 		}
```

Comment:
Using try/catch extends the time of the test significantly because it waits until the full timeout before it moves on, which means 30 seconds of delay in this case. Instead, I'd suggest using `Promise.any` so that it immediately picks up the first available scenario. For example:

```js
async selectSavedCard( cardHolderName: string ): Promise< void > {
	const cardSelector = this.page
		.locator( selectors.existingCreditCard( cardHolderName ) )
		.first();
	const editPaymentButton = this.page.locator( selectors.editPaymentStep );

	await Promise.any( [
		( async () => {
			await cardSelector.click();
		} )(),
		( async () => {
			await editPaymentButton.click();
			await cardSelector.click();
		} )(),
	] );
}
```

Ideally, we shouldn't need to apply these kinds of hacks, as the state should never be unpredictable like that. I'm aware it sometimes is unavoidable since we're testing in parallel against the same accounts, but in this case it would be really helpful to know where the 'edit' step will or will not be collapsed so that we understand why this is necessary.

Reply:
Thank you! I'm glad to see something exists. The only problem I found with the above is that apparently sometimes both buttons are visible, so I get this error:

```
    locator.waitFor: Error: strict mode violation: locator('label[for*="existingCard"]:has-text("End to End Testing")').first().or(locator('button[aria-label="Edit the payment method"]')) resolved to 2 elements
```

I believe I solved that by changing the line to use `first()` like so:

```
await cardSelector.or( editPaymentButton ).first().waitFor( { state: 'visible' } );
```

Hopefully I'm using that right. I've made https://github.com/Automattic/wp-calypso/pull/85663 to try this. Feel free to leave any further refinements there!